projects/igniteui-angular/src/lib/icon/icon.service.ts

```diff
@@ -48,14 +48,19 @@ export class IgxIconService {
     private _familyAliases = new Map<string, string>();
     private _cachedSvgIcons = new Map<string, Map<string, SafeHtml>>();
     private _iconLoaded = new Subject<IgxIconLoadedEvent>();
-    private _domParser = new DOMParser();
+    private _domParser: DOMParser;
 
     constructor(
         @Optional() private _sanitizer: DomSanitizer,
         @Optional() private _httpClient: HttpClient,
-        @Optional() @Inject(DOCUMENT) private _document: any
+        @Optional() @Inject(DOCUMENT) private _document: any,
+        @Inject(PLATFORM_ID) private platformId: any
     ) {
         this.iconLoaded = this._iconLoaded.asObservable();
+
+        if(isPlatformBrowser(this.platformId)) {
```

Comment:
Don't make the checks where the `cacheSvgIcon` is called. Make the check inside the first `if` in the `cacheSvgIcon` method instead. This way it would be more future proof if the method is called from somewhere else in the future.